tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/MongoDb/IBsonDocumentProxy.cs

```diff
@@ -1,23 +1,22 @@
-// <copyright file="BsonDocumentProxy.cs" company="Datadog">
+// <copyright file="IBsonDocumentProxy.cs" company="Datadog">
 // Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
 // This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
 // </copyright>
 
+using Datadog.Trace.DuckTyping;
+
 namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.MongoDb
 {
     /// <summary>
     /// MongoDB.Bson.BsonDocument interface for duck-typing
     /// </summary>
-    internal class BsonDocumentProxy
+    internal interface IBsonDocumentProxy : IDuckType
     {
         /// <summary>
         /// Gets an element of this document.
         /// </summary>
         /// <param name="index">The zero based index of the element.</param>
         /// <returns>The element.</returns>
-        public virtual BsonElementStruct GetElement(int index)
-        {
-            return default;
-        }
+        public BsonElementStruct GetElement(int index);
```

Comment:
The `BsonElement` this is ducktyping is a `struct`, so the return type here _must_ be non-nullable. Currently you get a weird runtime exception if you try to use this:

```csharp
                     //👇 Can't use this
public BsonElementStruct? GetElement(int index);
```

as it gives this:

```
Unable to access IWireProtocol.Command properties. Datadog.Trace.DuckTyping.DuckTypeDuckCopyStructDoesNotContainsAnyField: The [DuckCopy] struct 'System.Nullable`1[[Datadog.Trace.ClrProfiler.AutoInstrumentation.MongoDb.BsonElementStruct, Datadog.Trace, Version=2.41.0.0, Culture=neutral, PublicKeyToken=def86d061d0d2eeb]]' does not contains any public field. Remember that DuckCopy proxies must be declared using fields instead of properties.
```

It's incorrect to use `BsonElementStruct?` in this case, but the exception we throw is "wrong". I'll follow up separately so that we check for the `struct`/`Nullable<struct>` mismatch and throw an appropriate exception